[pytorch] pin dill version to 0.3.1

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31859 [DO NOT COMMIT] trigger docker image build job
* **#31870 [pytorch] pin dill version to 0.3.1**
* #31858 [pytorch] update llvmlite to 0.30.0

Summary:
Attempt to fix error on https://app.circleci.com/jobs/github/pytorch/pytorch/4135110
```
Traceback (most recent call last):
  File "test_torch.py", line 4191, in test_serialization_dill
    x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 597, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 761, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 270, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 467, in __init__
    StockUnpickler.__init__(self, *args, **kwds)
TypeError: __init__() got an unexpected keyword argument 'encoding'
```

ljk53 committed:
[pytorch] pin dill version to 0.3.1

Summary:
Attempt to fix error on https://app.circleci.com/jobs/github/pytorch/pytorch/4135110
```
Traceback (most recent call last):
  File "test_torch.py", line 4191, in test_serialization_dill
    x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 597, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 761, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 270, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 467, in __init__
    StockUnpickler.__init__(self, *args, **kwds)
TypeError: __init__() got an unexpected keyword argument 'encoding'
```

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] pin dill version to 0.3.1"


Summary:
Attempt to fix error on https://app.circleci.com/jobs/github/pytorch/pytorch/4135110
```
Traceback (most recent call last):
  File "test_torch.py", line 4191, in test_serialization_dill
    x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 597, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 761, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 270, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 467, in __init__
    StockUnpickler.__init__(self, *args, **kwds)
TypeError: __init__() got an unexpected keyword argument 'encoding'
```

[ghstack-poisoned]

ljk53 committed:
Update on "[pytorch] pin dill version to 0.3.1"


Summary:
Attempt to fix error on https://app.circleci.com/jobs/github/pytorch/pytorch/4135110
```
Traceback (most recent call last):
  File "test_torch.py", line 4191, in test_serialization_dill
    x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 597, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 761, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 270, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 467, in __init__
    StockUnpickler.__init__(self, *args, **kwds)
TypeError: __init__() got an unexpected keyword argument 'encoding'
```

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 86cefba3:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=bc68a8745f1838f4e519f2f456593f8189ba23ee&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base bc68a874 since Jan 09
  <details>
  <summary>
  You may want to rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=86cefba384b4641ef15af946569bea46a5d26d69).

### 1 failure *not* recognized by patterns:
| Job | Step | Status |
| --- | --- | --- |
| ![CircleCI](https://avatars0.githubusercontent.com/ml/7?s=12) <sup>pytorch_linux_xenial_cuda9_cudnn7_py3_build</sup> | <sup>Build</sup> | &#128721; [Broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=bc68a8745f1838f4e519f2f456593f8189ba23ee&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35&highlight_job=pytorch_linux_xenial_cuda9_cudnn7_py3_build) |


---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

ljk53 committed:
Update on "[pytorch] pin dill version to 0.3.1"


Summary:
Attempt to fix error on https://app.circleci.com/jobs/github/pytorch/pytorch/4135110
```
Traceback (most recent call last):
  File "test_torch.py", line 4191, in test_serialization_dill
    x2 = torch.load(f, pickle_module=dill, encoding='utf-8')
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 597, in load
    return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/torch/serialization.py", line 761, in _legacy_load
    magic_number = pickle_module.load(f, **pickle_load_args)
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 270, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/opt/python/2.7.9/lib/python2.7/site-packages/dill/_dill.py", line 467, in __init__
    StockUnpickler.__init__(self, *args, **kwds)
TypeError: __init__() got an unexpected keyword argument 'encoding'
```

[ghstack-poisoned]
